diff --git a/cmake/Modules/FindoneDNN.cmake b/cmake/Modules/FindoneDNN.cmake
index d91124f..011fb89 100644
--- a/cmake/Modules/FindoneDNN.cmake
+++ b/cmake/Modules/FindoneDNN.cmake
@@ -17,7 +17,15 @@ if(NOT ONEDNN_FOUND)
   if(POLICY CMP0169)
     cmake_policy(SET CMP0169 OLD)
   endif()
+  # --- Local oneDNN override for Nix ---
+  if(DEFINED FETCHCONTENT_SOURCE_DIR_ONEDNN)
+    message(STATUS "oneDNN: using local source at ${FETCHCONTENT_SOURCE_DIR_ONEDNN}")
 
+    set(onednn_SOURCE_DIR ${FETCHCONTENT_SOURCE_DIR_ONEDNN})
+    set(onednn_BINARY_DIR ${FETCHCONTENT_SOURCE_DIR_ONEDNN}/build)
+    set(onednn_POPULATED TRUE)
+
+  else()
   message(
     STATUS
       "oneDNN: fetching from ${ONEDNN_GIT_REPO} (commit: ${ONEDNN_GIT_TAG})")
@@ -27,6 +35,7 @@ if(NOT ONEDNN_FOUND)
     GIT_TAG ${ONEDNN_GIT_TAG}
     GIT_PROGRESS TRUE
     GIT_SHALLOW FALSE)
+  endif()
 
   set(DNNL_ENABLE_PRIMITIVE_CACHE
       TRUE
